src/TestData/resources/js/donations.js

```diff
@@ -0,0 +1,198 @@
+import API, { CancelToken } from './api';
+
+const { __, sprintf } = wp.i18n;
+const donationStatuses = document.querySelectorAll( 'input[name*="give_test_data_status"]' );
+const generateDonationsBtn = document.querySelector( '#give-test-data-generate-donations' );
+
+const handleStatusSelect = ( e ) => {
+	const element = e.currentTarget;
+	const status = element.dataset.status;
+
+	document
+		.querySelectorAll( `input[name*="give_test_data_count[${ status }]"], input[name*="give_test_data_revenue[${ status }]"]` )
+		.forEach( ( input ) => {
+			input.toggleAttribute( 'disabled', ! element.checked );
+		} );
+};
+
+/**
+ * @param  {HTMLElement} element Selected donation element
+ * @return {{revenue: number, statusName: string, count: number, status: string}} element state
+ */
+const getSelectedStatusState = ( element ) => {
+	const { status, name } = element.dataset;
+	return {
+		status,
+		statusName: name,
+		count: parseInt( document.querySelector( `input[name*="give_test_data_count[${ status }]"]` ).value ),
+		revenue: parseInt( document.querySelector( `input[name*="give_test_data_revenue[${ status }]"]` ).value ),
+	};
+};
+
+const processHasErrors = () => document.querySelector( '.give-test-data-process-error' );
```

Comment:
It would be better if we could track donations at the application level, as state, instead of relying on the DOM.